Zcs azzurro integration

AlePerla opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
New integration for ZCS Azzurro Inverters family equipped with a network datalogger (tested with original WiFi datalogger)

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- Link to documentation pull request: [home-assistant/home-assistant.io/pull/26298](https://github.com/home-assistant/home-assistant.io/pull/26298)

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

AlePerla committed:
working version

Lash-L commented on homeassistant/components/tplink_omada/translations/en.json:
```diff
@@ -1,39 +1,40 @@
 {
```
Seems like you accidentally committed this change. pre-commit auto changed that file.

AlePerla replied:
Yes I probably got confused by the name, either it has been a merge error. It should be fixed now

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/__init__.py:
```diff
@@ -0,0 +1,51 @@
+from .const import DOMAIN, SCHEMA_CLIENT_KEY, SCHEMA_FRIENDLY_NAME, SCHEMA_THINGS_KEY
+from .zcs_azzurro_api import ZcsAzzurroApi
+
+_LOGGER = logging.getLogger(__name__)
```
logger is unused in this file

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/__init__.py:
```diff
@@ -0,0 +1,51 @@
+    """Set up zcs_azzurro from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = ZcsAzzurroApi(
+        client=entry.data[SCHEMA_CLIENT_KEY],
+        thing_serial=entry.data[SCHEMA_THINGS_KEY],
+        name=entry.data[SCHEMA_FRIENDLY_NAME],
+    )
```
hass.data should only be set once you can confirm that you can authenticate and connect. You probably should also be using ConfigEntryNotReady and ConfigEntryAuthFailed Exceptions

AlePerla replied:
changed. Same as above, scaffold put it here

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/config_flow.py:
```diff
@@ -0,0 +1,132 @@
+            SCHEMA_THINGS_KEY: "Thing serial",
+            SCHEMA_FRIENDLY_NAME: "Friendly name",
+        }
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            _LOGGER.debug(
+                "before create entity info is %s and data is %s", info, user_input
+            )
+            return self.async_create_entry(
+                title=info["title"],
+                data=user_input,
+                description_placeholders=placeholders,
+            )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=STEP_USER_DATA_SCHEMA,
+            errors=errors,
+            description_placeholders=placeholders,
+        )
```
If you change this to user_input is not None: then you don't need to call async_show_form to begin with.

Just handle_user_input if exist
show_form

Instead of 
show form
handle user input
show form

AlePerla replied:
Right, good suggestion. The script.scaffold integration command makes it this way though, perhaps it needs to be changed as well

Lash-L replied:
Yep. I submitted a PR for the scaffold to be changed, it should be merged shortly.

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/manifest.json:
```diff
@@ -0,0 +1,15 @@
+  "config_flow": true,
+  "documentation": "https://www.home-assistant.io/integrations/zcs_azzurro",
+  "requirements": ["requests==2.28.1"],
+  "ssdp": [],
+  "zeroconf": [],
+  "homekit": {},
+  "dependencies": [],
```
You can remove the unused ones

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/manifest.json:
```diff
@@ -0,0 +1,15 @@
+{
+  "domain": "zcs_azzurro_inverter",
+  "name": "ZCS Azzurro inverter",
+  "config_flow": true,
+  "documentation": "https://www.home-assistant.io/integrations/zcs_azzurro",
+  "requirements": ["requests==2.28.1"],
+  "ssdp": [],
+  "zeroconf": [],
+  "homekit": {},
+  "dependencies": [],
+  "codeowners": ["@AlePerla"],
+  "iot_class": "cloud_polling",
+  "integration_type": "hub",
+  "loggers": ["zcs_azzurro_inveter"]
```
Also I believe this is now supposed to be in domain, name, then alphabetical order.

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/strings.json:
```diff
@@ -0,0 +1,22 @@
+      "unknown": "[%key:common::config_flow::error::unknown%]"
+    },
+    "abort": {
+      "already_configured": "[%key:common::config_flow::abort::already_configured_device%]"
```
You don't seem to check do in config_flow. You probably should

AlePerla replied:
Sorry, I did not get what I should check in config_flow. These are there because of scaffold as well, I did not change anything

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/zcs_azzurro_api.py:
```diff
@@ -0,0 +1,137 @@
+            raise ConnectionError("Response did not return correctly")
+        response_data: dict[str, Any] = response.json()[ZcsAzzurroApi.REALTIME_DATA_KEY]
+        _LOGGER.debug("fetched realtime data %s", response_data)
+        if not response_data[ZcsAzzurroApi.RESPONSE_SUCCESS_KEY]:
+            raise ConnectionError("Response did not return correctly")
+        return response_data[ZcsAzzurroApi.PARAMS_KEY][
+            ZcsAzzurroApi.RESPONSE_VALUES_KEY
+        ][0][self._thing_serial]
+
+    def alarms_request(self) -> dict:
+        """Request alarms."""
+        required_values = [ZcsAzzurroApi.REQUIRED_VALUES_ALL]
+        data = {
+            ZcsAzzurroApi.DEVICES_ALARMS_KEY: {
+                ZcsAzzurroApi.COMMAND_KEY: ZcsAzzurroApi.DEVICES_ALARMS_COMMAND,
+                ZcsAzzurroApi.PARAMS_KEY: {
+                    ZcsAzzurroApi.PARAMS_THING_KEY: self._thing_serial,
+                    ZcsAzzurroApi.PARAMS_REQUIRED_VALUES_KEY: ZcsAzzurroApi.REQUIRED_VALUES_SEP.join(
+                        required_values
+                    ),
+                },
+            }
+        }
+        response = self._post_request(data)
+        if not response.ok:
+            raise ConnectionError("Response did not return correctly")
+        response_data: dict[str, Any] = response.json()[
+            ZcsAzzurroApi.DEVICES_ALARMS_KEY
+        ]
+        _LOGGER.debug("fetched realtime data %s", response_data)
+        if not response_data[ZcsAzzurroApi.RESPONSE_SUCCESS_KEY]:
+            raise ConnectionError("Response did not return correctly")
+        return response_data[ZcsAzzurroApi.PARAMS_KEY][
+            ZcsAzzurroApi.RESPONSE_VALUES_KEY
+        ][0][self._thing_serial]
+
+    @property
+    def identifier(self):
+        """object identifier."""
+        return f"{self.client}_{self._thing_serial}"
```
You need to make this an external package.
https://developers.home-assistant.io/docs/creating_component_code_review?_highlight=pypi#4-communication-with-devicesservices

AlePerla replied:
I thought it was a [strong] suggestion, not a must (but I noticed I was wrong, reading it again from your link). Either it is just few calls. It will require me a little longer to fix this

Lash-L reviewed (changes requested):
Here's some feedback! A lot of this is feedback I got from a PR on my integration, so I figured I would pass it along.

Lash-L commented:
Also I think you messed up while trying to catch your branch up with develop. Try merging upstream/develop into your branch.

AlePerla commented:
> Also I think you messed up while trying to catch your branch up with develop. Try merging upstream/develop into your branch.

You are right, I merged the dev branch on my fork. My bad sorry.
I get these errors on a file I did not change though:
homeassistant/components/recorder/statistics.py:127: error: Unused "type: ignore" comment
homeassistant/components/recorder/statistics.py:379: error: Returning Any from function declared to return "int"  [no-any-return]
homeassistant/components/recorder/statistics.py:857: error: Unused "type: ignore" comment
homeassistant/components/recorder/statistics.py:1260: error: Unused "type: ignore" comment
homeassistant/components/recorder/statistics.py:1306: error: Unused "type: ignore" comment

dtrodrigues committed:
Fix typo in requires-python hassfest error message (#87896)

starkillerOG committed:
Bump reolink-aio to 0.4.1 (#87933)

emontnemery committed:
Bump py-dormakaba-dkey to 1.0.2 (#87935)

bdraco committed:
Speed up selecting which statistics to compile (#87938)

zxdavb committed:
Bump incomfort client library to 0.5.0 (#87877)

allenporter committed:
Improve Google Calendar config flow error message when API disabled (#87889)

bdraco committed:
Speed up loops with jinja templates (#87945)

klaasnicolaas committed:
Bump gridnet to v4.2.0 (#87903)

emontnemery committed:
Enable strict typing for the dormakaba_dkey integration (#87934)

epenet committed:
Add type hints to integration tests (part 5) (#87850)

rappenze committed:
Small code improvements fibaro integration (#87949)

bdraco committed:
Small cleanups to template helper (#87944)

- reduce dict lookups
- avoid split_entity_id

muppet3000 committed:
Update growatt_server owners (#87804)

bdraco committed:
Speed up template lru_caches (#87942)

By avoiding the argument unpacking these functions are faster
and reduce stack overhead

janiversen committed:
Bump pymodbus to v3.1.3 (#87841)

mampfes committed:
Bump wiffi to 1.1.2 (#87798)

bdraco committed:
Make mqtt discovery use callback functions (#87883)

bdraco committed:
Make mqtt websocket subscribe use callback functions (#87885)

Nothing was being awaited so this was needlessly generating tasks when the callbacks fired

Ernst79 committed:
Fix kegtron state class for volume sensors (#87950)

bdraco committed:
Optimize mqtt matchers (#87853)

bdraco committed:
Avoid creating a task when waiting for the MQTT mid (#87887)

jbouwh committed:
Improve MQTT `Callable` type hints (#87838)

bdraco committed:
Use PidfdChildWatcher by default when available (#87951)

This is a backport from cpython 3.12

https://docs.python.org/3/library/asyncio-policy.html
> PidfdChildWatcher is a “Goldilocks” child watcher implementation. It doesn’t require signals or threads, doesn’t interfere with any processes launched outside the event loop, and scales linearly with the number of subprocesses launched by the event loop. The main disadvantage is that pidfds are specific to Linux, and only work on recent (5.3+) kernels.

https://github.com/python/cpython/pull/98024
There are some additional fixes in cpython 3.12
in https://github.com/python/cpython/pull/94184
when there is no event loop running in the main thread
but this is not a problem we have

Ernst79 committed:
Bump bthome-ble to 2.5.2 (#87862)

starkillerOG committed:
Netgear allow to remove old devices (#87849)

jbouwh committed:
Complete partly implemented type hints for `dict` on MQTT platforms (#87839)

rappenze committed:
Remove deprecated fibaro yaml config (#87939)

bdraco committed:
The recorder/statistic(s)_during_period websocket API now require the statistic_id(s) field (#87937)

raman325 committed:
Bump zwave-js-server-python to 0.45.2 (#87822)

Lash-L committed:
Rework user_input check in config flow scaffold (#87820)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

gjohansson-ST committed:
Change name to entity_id update platform error messages (#87960)

racsa1 committed:
Bump pydaikin 2.9.0 (#87067)

Co-authored-by: rsa <>

gjohansson-ST committed:
Change name to entity_id Number platform error messages (#87961)

gjohansson-ST committed:
Change name to entity_id Select platform error messages (#87962)

ludeeus committed:
Add recorder engine name and version to analytics (#87784)

* Add recorder to analytics

* Add test

* Add recorder to after_dependencies

* Add version

* dialect can not be None

epenet committed:
Add type hints to integration tests (part 6) (#87979)

epenet committed:
Add type hints to integration tests (part 7) (#87980)

emontnemery committed:
Fix sensor unit conversion (#87978)

epenet committed:
Add type hints to integration tests (part 10) (#87985)

emontnemery committed:
Add dormakaba_dkey battery sensor (#87975)

* Add dormakaba_dkey battery sensor

* Update .coveragerc

* Apply suggestion from code review

* Update .coveragerc

epenet committed:
Add type hints to integration tests (part 8) (#87982)

starkillerOG committed:
Reolink late review comments (#87991)

late review comments

frenck committed:
Add more entities for Elgato Key Light Mini (#87793)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

epenet committed:
Add type hints to integration tests (part 9) (#87983)

frenck committed:
Add homeassistant.reload_all service (#87769)

zxdavb committed:
Change geniushub codeowner (#87918)

jbouwh committed:
Correct typing on hass_storage fixure (#87986)

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

frenck committed:
Update pytest-asyncio to 0.20.3 (#87995)

mletenay committed:
Add support for 4 MPTT inverter (#86901)

gjohansson-ST committed:
Change name to entity_id Text platform error messages (#87963)

epenet committed:
Add type hints to integration tests (part 12) (#87997)

emontnemery committed:
Adjust suggested_display_precision according to unit conversion (#87614)

epenet committed:
Add type hints to integration tests (part 11) (#87996)

balloob committed:
Make all history stats tests async (#87973)

epenet committed:
Add type hints to integration tests (part 13) (#87998)

epenet committed:
Add mariadb 10.6.10 to database tests (#87989)

bdraco committed:
Avoid subprocess memory copy when c library supports posix_spawn (#87958)

* use posix spawn on alpine

* Avoid subprocess memory copy when c library supports posix_spawn

By default python 3.10 will use the fork() which has to
copy all the memory of the parent process (in our case
this can be huge since Home Assistant core can use
hundreds of megabytes of RAM). By using posix_spawn
this is avoided.

In python 3.11 vfork will also be available
https://github.com/python/cpython/issues/80004#issuecomment-1093810689
https://github.com/python/cpython/pull/11671 but we won't
always be able to use it and posix_spawn is considered safer
https://bugzilla.kernel.org/show_bug.cgi?id=215813#c14

The subprocess library doesn't know about musl though
even though it supports posix_spawn https://git.musl-libc.org/cgit/musl/log/src/process/posix_spawn.c
so we have to teach it since it only has checks for glibc
https://github.com/python/cpython/blob/1b736838e6ae1b4ef42cdd27c2708face908f92c/Lib/subprocess.py#L745

The constant is documented as being able to be flipped here:
https://docs.python.org/3/library/subprocess.html#disabling-use-of-vfork-or-posix-spawn

* Avoid subprocess memory copy when c library supports posix_spawn

By default python 3.10 will use the fork() which has to
copy memory of the parent process (in our case
this can be huge since Home Assistant core can use
hundreds of megabytes of RAM). By using posix_spawn
this is avoided and subprocess creation does not
get discernibly slow the larger the Home Assistant
python process grows.

In python 3.11 vfork will also be available
https://github.com/python/cpython/issues/80004#issuecomment-1093810689
https://github.com/python/cpython/pull/11671 but we won't
always be able to use it and posix_spawn is considered safer
https://bugzilla.kernel.org/show_bug.cgi?id=215813#c14

The subprocess library doesn't know about musl though
even though it supports posix_spawn https://git.musl-libc.org/cgit/musl/log/src/process/posix_spawn.c
so we have to teach it since it only has checks for glibc
https://github.com/python/cpython/blob/1b736838e6ae1b4ef42cdd27c2708face908f92c/Lib/subprocess.py#L745

The constant is documented as being able to be flipped here:
https://docs.python.org/3/library/subprocess.html#disabling-use-of-vfork-or-posix-spawn

* missed some

* adjust more tests

* coverage

bdraco committed:
Bump aioesphomeapi to 13.3.1 (#87969)

* Bump aioesphomeapi to 13.3.0

We probably need to include https://github.com/esphome/aioesphomeapi/pull/382
as well in another bump

* bump

MartinHjelmare committed:
Refactor tradfri fan tests (#87999)

MartinHjelmare committed:
Refactor tradfri diagnostics tests (#88017)

bdraco committed:
Bump aiohttp to 3.8.4 (#88032)

emontnemery committed:
Increase test scope of MariaDB + PostgreSQL tests (#87019)

Co-authored-by: J. Nick Koston <nick@koston.org>

shmuelzon committed:
Add mjpeg camera stream_source (#86908)

Co-authored-by: J. Nick Koston <nick@koston.org>

epenet committed:
Add mariadb 10.3.32 to database tests (#87741)

Co-authored-by: J. Nick Koston <nick@koston.org>

Danielhiversen committed:
Update tibber lib 0.26.13 (#88018)

bachya committed:
Bump `pyopenuv` to 2023.02.0 (#88039)

dcmeglio committed:
Add support for telnet connections for Denonavr integration (#85980)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Fix new stats rows creating data in the old created column (#88013)

bdraco committed:
Restore original aiohttp cancelation behavior (#88046)

balloob committed:
Keep task references while running (#87970)

* Keep task references while running

* Update pilight tests pointing at correct logger call

* Fix graphite tests

* Fix profiler tests

* More graphite test fixes

* Remove extra sleep

* Fix tests

* Shutdown background tasks as part of stage 1

* Remove unnecessary sleep in test

* Remove unused method on mock hass

* Skip on cancelled too

* Remove background tasks

* Test trigger variables without actually sleeping

* Fix graphite

* One more graphite grrrrrrr

bdraco committed:
Bump zeroconf to 0.47.3 (#88045)

changelog: https://github.com/python-zeroconf/python-zeroconf/compare/0.47.1...0.47.3

bdraco committed:
Bump dbus-fast to 1.84.1 (#88042)

bdraco committed:
Bump home-assistant-bluetooth to 1.9.3 (#88051)

mkmer committed:
Bump AIOAladdinConnect 0.1.56 (#88041)

rfleming71 committed:
Bump pyoctoprintapi to 0.1.11 (#88052)

Bump pyoctoprint to get camera url fixes

balloob committed:
Handle device reg fields not being valid data in openai conversion (#88047)

Handle device reg fields not being valid data

emontnemery committed:
Add dormakaba_dkey door and dead bolt binary sensors (#87990)

* Add dormakaba_dkey door and dead bolt binary sensors

* Rename dead bolt sensor

* Fix docstring

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

epenet committed:
Cleanup renault services (#88064)

farmio committed:
Update xknx to 2.5.0 (#88069)

MartinHjelmare committed:
Refactor tradfri sensor tests (#88026)

bachya committed:
Add a calendar entity to Ridwell (#86117)

Danielhiversen committed:
Fix Tibber stats (#88025)

bdraco committed:
Change DeviceAutomationDetails to use a dataclass for py3.11 (#88087)

A NamedTuple can no longer be an enum value in py3.11

bdraco committed:
Fix group integration with py3.11 (#88089)

In py3.11 `Passing coroutines is forbidden, use tasks explicitly`

MartinHjelmare committed:
Fix apple tv untracked task (#88102)

rytilahti committed:
Shorten tplink update cycle (#87399)

bdraco committed:
Fix unifiprotect with python 3.11 (#88101)

There are some breaking changes to Enum https://blog.pecar.me/python-enum

needs https://github.com/AngellusMortis/pyunifiprotect/pull/263 but
does not need to be in this PR

bdraco committed:
Fix sleepiq with python 3.11 (#88098)

There is a breaking change in enum behavior

see https://blog.pecar.me/python-enum

bdraco committed:
Fix invalid spec in mqtt tests for py3.11 (#88092)

Fix invalid spec in mqtt tests

unittest.mock.InvalidSpecError: Cannot spec a Mock object.

rytilahti committed:
Prefer effect over other light settings for tplink (#85642)

bdraco committed:
Fix forked_daapd integration with py3.11 (#88091)

In py3.11 `Passing coroutines is forbidden, use tasks explicitly`

bdraco committed:
Fix hassio integration with py3.11 (#88090)

In py3.11 `Passing coroutines is forbidden, use tasks explicitly`

bdraco committed:
Fix cloud integration with py3.11 (#88088)

In py3.11 `Passing coroutines is forbidden, use tasks explicitly`

bdraco committed:
Fix rmvtransport integration with py3.11 (#88094)

In py3.11 `Passing coroutines is forbidden, use tasks explicitly`

bdraco committed:
Add strict typing to yalexs_ble (#88086)

* Add strict typing to yalexs_ble

* Add strict typing to yalexs_ble

* Add strict typing to yalexs_ble

bachya committed:
Revert "Add a calendar entity to Ridwell (#86117)" (#88106)

This reverts commit 16a52754611a14baaff98d92abe72afbc3ec4e9f.

Kane610 committed:
Bump pydeconz to v107 (#88114)

MartinHjelmare committed:
Fix august untracked task (#88107)

bdraco committed:
Fix zwave_me tests with python 3.11 (#88122)

bdraco committed:
Fix vera tests with python3.11 (#88124)

bdraco committed:
Fix wemo tests with py3.11 (#88123)

balloob committed:
Graphite flaky test fix (#88115)

bdraco committed:
Fix apprise tests for python 3.11 (#88126)

bdraco committed:
Fix twentemilieu diagnostics with python 3.11 (#88121)

mgjbroadbent committed:
Update orjson to resolve segmentation fault during JSON serialisation (#88119)

Home Assistant uses orjson 3.8.5 that contains an issue[1] on musl libc
platforms that causes a segmentation fault.  This particularly affect
Home Assistant container installations reported in #87283 and #87522.

This updates the version to 3.8.6 that resolves the segmentation
fault during json serialisation.

[1] https://github.com/ijl/orjson/issues/335

elupus committed:
Bump arcam dependency to 1.2.1 (#88125)

jbouwh committed:
Fix race on check MQTT subscriptions (#88117)

* Check MQTT subscriptions under paho client lock

* Check simple subscriptions first

* Apply suggestion

sw-carlos-cristobal committed:
Replace Fitbit weight SensorStateClass measurement with total (#88118)

* Remove SensorStateClass.MEASUREMENT from weight

* Using SensorStateClass.TOTAL for weight

bdraco committed:
Fix exception ordering in yeelight for python 3.11 (#88134)

bdraco committed:
Bump pyunifiprotect to 4.7.0 for python 3.11 (#88130)

bdraco committed:
Fix enum try_parse test for python3.11 (#88131)

dcmeglio committed:
Move denonavr state update callbacks to async_added_to_hass (#88147)

Move state update callbacks to async_added_to_hass.

bdr99 committed:
Fix typo in OpenAI prompt (#88137)

bdraco committed:
Bump docker base to 2023.02.0 (#88084)

* Bump base image to 2023.02.0

changelog: https://github.com/home-assistant/docker-base/compare/2022.11.0...2023.02.0

* Bump base image to 2023.02.0

changelog: https://github.com/home-assistant/docker-base/compare/2022.11.0...2023.02.0

jbouwh committed:
Fix doc string in mqtt test (#88138)

Fix docstr

epenet committed:
Add type hints to integration tests (part 15) (#88006)

bieniu committed:
Add support for `triple_push` event for Shelly gen2 devices (#88081)

* Add support for triple_push event

* Sort

* Fix tests

bieniu committed:
Do not reload Shelly TRV entry when not needed (#88037)

* Do not reload the entry when the target temperature changes

* Do not reload the entry when the mode changes

* Increase test coverage

* Increase test coverage

ThomDietrich committed:
Statistics component fix device_class for incremental source sensors (#88096)

* Return None device_class for incremental source sensors

* Ignore linting error

* Fix ignore linting error

* Fix ignore linting error

* Fix ignore linting error

* Catch potential parsing error with enum

epenet committed:
Add type hints to integration tests (part 14) (#88005)

epenet committed:
Fix incorrect reference to json WriteError (#88161)

jbouwh committed:
Add note on always subscribing to the MQTT broker when topics are shared (#88159)

Add note on always subscribing

epenet committed:
Add type hints to integration tests (part 16) (#88008)

mheath committed:
Support Fully Kiosk Browser device discovery via MQTT (#88109)

Support FKB device discovery via MQTT

epenet committed:
Add type hints to integration tests (part 17) (#88163)

frenck committed:
Enable Ruff SIM118 (#87772)

epenet committed:
Fix mobile app regression accepting sensor states (#88164)

* Fix mobile app regression accepting sensor states

* Adjust tests

bdraco committed:
Fix netatmo with python 3.11 (#88093)

frenck committed:
Enable Ruff PT006 (#88165)

* Enable Ruff PT006

* Adjust existing cases

* Fix tests

* Remove unneeded parentheses

epenet committed:
Remove invalid pragma: no cover (#88173)

frenck committed:
Enable Ruff SIM201 (#88171)

epenet committed:
Expose DEVICE_CLASS_STATE_CLASSES sensor constant (#88172)

epenet committed:
Adjust imports in rflink test_init (#88177)

emontnemery committed:
Add WS command for discovering Thread routers (#88021)

* Add WS command for discovering Thread routers

* Fix type annotations

* Mock zeroconf in tests

* Key discovery by external MAC address

* Add tests

* Include hostname in data, allow missing fields

* Fix typo

* Include server instead of hostname

starkillerOG committed:
Bump reolink-aio to 0.4.2 (#88175)

frenck committed:
Enable Ruff SIM212 (#88181)

epenet committed:
Add type hints to integration tests (part 20) (#88179)

frenck committed:
Remove uamqp version constrain (#88176)

riokuu committed:
Blebox fix thermobox reporting wrong state (#88169)

fix: fixed climate hvac_mode for when device is off

scop committed:
Do not round Huawei LTE LTE frequency values (#88127)

Suggest display precision 0 instead.

While at it, fix source info in frequency formatter docstring.

gertjanstulp committed:
Fixed float number validation in sensor component (#88074)

epenet committed:
Add postgresql 12.14 to database tests (#88071)

* Add postgre 12.14 to database tests

* Rename jobs

mkmer committed:
Honeywell disable detergent level by default (#88040)

* Disable fill by default

* Fix tests

* use TANK_FILL.get

* Remove None from attribute get
add reload to sensor test

* Typing
fix iteration error

emontnemery committed:
Fix otbr tests (#88184)

bdraco committed:
Disable profiler.memory service with python 3.11 (#88136)

jbouwh committed:
Allow resetting MQTT climate attributes and ignore empty values (#87936)

Allow reseting MQTT climate attributes

bdraco committed:
Increase database test timeout (#88187)

Jc2k committed:
Support HomeKit Controller Thread Provisioning (#87809)

epenet committed:
Add type hints to integration tests (part 19) (#88178)

bdraco committed:
Account for changed python3.11 enum.IntFlag 0 behavior in zha (#88144)

emontnemery committed:
Add WS command thread/delete_dataset (#88077)

* Add WS command thread/delete_dataset

* Return not_allowed when trying to delete the preferred dataset

bdraco committed:
Fix aprs tests with python 3.11 (#88189)

* Fix aprs tests with python 3.11

* hints

marcelveldt committed:
Bump python-matter-server to 2.1.0 (#88192)

* Bump python-matter-server to 2.1.0

* Fix tests

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

scop committed:
Let black infer target-version (#88199)

frenck committed:
Constrain uamqp for Python 3.10 only (#88198)

frenck committed:
Update Ruff to v0.0.247 (#88210)

* Update Ruff to v0.0.247

* Fix new found violations

eifinger committed:
Bump voluptuous-serialize to 2.6.0 (#88211)

rappenze committed:
Update pyfibaro to version 0.6.8 (#88203)

Update pyfibaro library to version 0.6.8

bdraco committed:
Fix allowing identical flows to be created before startup (#88213)

The check for identical flows only worked after
the start event. We now check against pending
flows as well

If startup took a while we could end up
with quite the thundering herd

bdraco committed:
Refactor zeroconf task handling (#88204)

* Refactor zeroconf task handling

- Avoid the need to create tasks for most callbacks
- Fixes the untracked task that could get unexpectedly GCed

* be consistant

* be consistant

* fix zeroconf tests

* runtime

* Revert "runtime"

This reverts commit 19e6b6183759279490efbed3129dffd447dc4808.

* precalc

* refactor

* tweak

* update tests

bdr99 committed:
Fix capitalization of OwnTone integration (#88219)

epenet committed:
Add type hints to integration tests (part 21) (#88233)

frenck committed:
Update typing-extensions to 4.5.0 (#88236)

epenet committed:
Inverse json import logic (#88099)

* Fix helpers and util

* Adjust components

* Move back errors

* Add report

* mypy

* mypy

* Assert deprecation messages

* Move test_json_loads_object

* Adjust tests

* Fix rebase

* Adjust pylint plugin

* Fix plugin

* Adjust references

* Adjust backup tests

emontnemery committed:
Drop unnecessary copy from entitydescription.name to _attr_name (#88251)

cdce8p committed:
Update pydantic to 1.10.5 (#88250)

epenet committed:
Add type hints to integration tests (part 22) (#88234)

bdraco committed:
Bump vulcan-api to 2.3.0 for python 3.11 (#88254)

Bump vulcan-api to 2.3.0

see https://github.com/kapi2289/vulcan-api/pull/126
see https://github.com/home-assistant/core/pull/88038
see https://github.com/home-assistant/docker/pull/260

marcelveldt committed:
Bump python-matter-server to 2.1.1 (#88258)

bdraco committed:
Fix restore of MQTT subscriptions from reload (#88220)

eifinger committed:
Fix default value for departure time in google_travel_time (#88255)

Fix default value for departure time

bdraco committed:
Speed up formatting statistics data (#88228)

* Speed up fetching statistics by using attrgetter

Uses the faster native code vs a lambda
https://github.com/python/cpython/blob/4aeae286715a3a9fa624429733582917606000c3/Modules/_operator.c#L1406

* avoid dict lookups

* avoid dict lookups

* naming

* split

* Revert "split"

This reverts commit 0ead89603d1570554148f580948e9bce9eca6d38.

* Revert "Revert "split""

This reverts commit 20014af16862d067b7f807f1eb0f909e89fd4b6a.

* tweak

* tweak

* tweak

* tweak

* Revert "tweak"

This reverts commit d5e4aac7c5cfc22fdabb016cefb9da6510578cd3.

* Revert "tweak"

This reverts commit 0cebae33f817e0a075dcb3f11d9c479cdc50d3dd.

* Revert "tweak"

This reverts commit 0184e47e2473b9855513bd45b601247f1d8be85e.

* comment

* comment

* comment

* reduce local vars

* less

bdraco committed:
Add python 3.11 to the CI (#88038)

* Remove profiler.memory service

guppy3 is not python3.11 compat
https://github.com/zhuyifei1999/guppy3/issues/41

This service will return if and when guppy3 becomes
python3.11 compat

* squash

* temp remove

* temp dump tests

* temp dump tests

* drop a few more to get a run

* drop a few more to get a run

* Account for changed python3.11 enum.IntFlag behavior in zha

There may be additional changes needed, but I could only
see what needed to be updated based on the tests

* merge

* restore

* restore

* legacy value

* tweak a bit for the python 3.11 timings

* block cchardet

* conditional

* adjust est

* test

* not yet

* tweak

* give a little leeway for timing

* Fix otbr tests

* Increase database test timeout

It looks like we need a little more time to run
with the addiitonal tests in #87019

* Increase database test timeout

It looks like we need a little more time to run
with the addiitonal tests in #87019

* Fix aprs tests with python 3.11

* merge fix

* hints

* Update homeassistant/package_constraints.txt

* Update script/gen_requirements_all.py

* Constrain uamqp for Python 3.10 only

* Bump vulcan-api to 2.3.0

see https://github.com/kapi2289/vulcan-api/pull/126
see https://github.com/home-assistant/core/pull/88038
see https://github.com/home-assistant/docker/pull/260

* add ban

* Bump python-matter-server to 2.1.1

* revert

* Update tests/asyncio_legacy.py

---------

Co-authored-by: Erik <erik@montnemery.com>
Co-authored-by: Franck Nijhof <git@frenck.dev>
Co-authored-by: Marcel van der Veldt <m.vanderveldt@outlook.com>

jbouwh committed:
Replace deprecated SSLContext constant PROTOCOL_TLS in mqtt (#88214)

Replace deprecated SSLContext constants

balloob committed:
Stop processing when we hit bad encryption in mobile app (#88150)

* Stop processing when we hit bad encryption

* Accept webhook payload that is a list

* Rename functions because we import them

* Revert a debug thing

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

epenet committed:
Add JSON support to load_fixture (#88076)

* Add JSON support to load_fixture

* More tests

* Remove lru_cache on load_json

mkmer committed:
Bump aiosomecomfort to 0.0.8 (#88195)

* Update is_aux_heat in runtime

* Fix merge mistake

* Merge fixes 2.0

* Bump aiosomecomfort to 0.0.8

bdraco committed:
Speed up compiling hourly statistics (#88225)

synesthesiam committed:
Use blocking in intent service calls and verify results (#88035)

* Use blocking in service calls and verify result

* Block for 2 seconds and update states after

* Small timeout in service call to allow exceptions

* Move sun test

epenet committed:
Refactor json tests to align with new code (#88247)

* Refactor json tests to align with new code

* Use tmp_path

Lash-L commented:
@AlePerla I think those should go away if you just merge upstream/dev into your branch. Not 100% sure though. You may have to do a force push

AlePerla commented:
@Lash-L I am about to publish the API on PyPi and then I will give it a try.
If it really gets too messy, perhaps I could put the change on another branch created by upstream/dev, the change is not so big, it wouldn't take much.
> You don't seem to check do in config_flow. You probably should
Could you explain me this? I did not get the requested modification

bdraco committed:
Switch yalexs_ble to use no_devices_found (#88267)

bdraco committed:
Switch switchbot to use no_devices_found (#88268)

bdraco committed:
Switch led_ble to use no_devices_found (#88269)

bdraco committed:
Switch keymitt_ble to use no_devices_found (#88271)

* Switch keymitt_ble to use no_devices_found

* Switch keymitt_ble to use no_devices_found

bdraco committed:
Switch ld2410_ble to use no_devices_found (#88270)

balloob committed:
Add support for background tasks in HA (#88265)

* Add support for background tasks

* make name mandatory for background tasks

* Update docstring

* async_create_background_task

* Grammar

jesserockz committed:
Implement Switchbot Blind Tilt (#86591)

Co-authored-by: J. Nick Koston <nick@koston.org>
Co-authored-by: BelowAverageDeveloper <90269575+BelowAverageDev@users.noreply.github.com>

balloob committed:
Fix dangling task for unifiprotect (#88300)

balloob committed:
Fix dangling task for zha (#88302)

balloob committed:
Fix dangling task for sonos (#88298)

balloob committed:
Fix dangling task for homekit (#88289)

bdraco committed:
Cleanup stale recorder code (#88275)

- Removes dead commit function
- Small cleanup to debug handling in execute

bdraco committed:
Fix handling of dns resolution failure during sense setup (#88279)

```
2023-02-16 20:44:54.516 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry alexander for sense
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 1152, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 1152, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 861, in _resolve_host
    await event.wait()
  File "/usr/local/lib/python3.10/site-packages/aiohttp/locks.py", line 34, in wait
    raise self._exc
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 874, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/resolver.py", line 33, in resolve
    infos = await self._loop.getaddrinfo(
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 860, in getaddrinfo
    return await self.run_in_executor(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/socket.py", line 955, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Try again

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 381, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/usr/src/homeassistant/homeassistant/components/sense/__init__.py", line 83, in async_setup_entry
    await gateway.get_monitor_data()
  File "/usr/local/lib/python3.10/site-packages/sense_energy/asyncsenseable.py", line 214, in get_monitor_data
    json = await self._api_call("app/monitors/%s/overview" % self.sense_monitor_id)
  File "/usr/local/lib/python3.10/site-packages/sense_energy/asyncsenseable.py", line 174, in _api_call
    async with self._client_session.get(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/connector.py", line 1166, in _create_direct_connection
    raise ClientConnectorError(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host api.sense.com:443 ssl:default [Try again]  ```

balloob committed:
Fix dangling task for elkm1 (#88286)

* Fix dangling task for elkm1

* Update homeassistant/components/elkm1/__init__.py

Co-authored-by: J. Nick Koston <nick@koston.org>

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

balloob committed:
Fix dangling task for homeassistant integration (#88288)

balloob committed:
Fix dangling task for google (#88287)

bdraco committed:
Bump ha-ffmpeg to 3.1.0 (#88304)

changelog: https://github.com/home-assistant-libs/ha-ffmpeg/compare/3.0.2...3.1.0

emontnemery committed:
Fix sleepiq test to not contact the API (#88315)

* Fix sleepiq test to not contact the API

* Add an autosue fixture for config entry setup

epenet committed:
Adjust async_setup_entry in config_flow scaffold (#88319)

epenet committed:
Fix incorrect condition for atmospheric_pressure (#88320)

emontnemery committed:
Fix recorder platform tests (#88330)

balloob committed:
Fix dangling task for wiz (#88301)

epenet committed:
Add type hints to integration tests (part 24) (#88307)

epenet committed:
Add type hints to integration tests (part 23) (#88235)

balloob committed:
Fix dangling task for roon (#88291)

Ernst79 committed:
Bump xiaomi-ble to 0.16.4 (#88333)

epenet committed:
Add type hints to integration tests (part 25) (#88308)

dcmeglio committed:
Add additional characteristics for Airversa Homekit Air Purifiers (#85660)

balloob committed:
Fix dangling task for cast (#88285)

epenet committed:
Add type hints to integration tests (part 18) (#88174)

epenet committed:
Add type hints to integration tests (recorder) (#88313)

epenet committed:
Add type hints to integration tests (zha) (#88309)

jjlawren committed:
Detect when battery-powered Sonos devices are turned off (#87606)

epenet committed:
Adjust json imports in tests (#88248)

bdraco committed:
Cleanup august binary sensors (#88332)

emontnemery committed:
Remove conditional test of self.hass from tplink_omada switch (#88325)

emontnemery committed:
Fail tests which cause an SQLAlchemy warning (#88327)

Co-authored-by: J. Nick Koston <nick@koston.org>

balloob committed:
Add background tasks to config entries (#88335)

* Use a set for config entries task tracking

* Allow adding background tasks to config entries

* Add tests for config entry add tasks

* Update docstrings on core create task

* Migrate roon and august

* Use in more places

* Guard for None

frenck committed:
Update pyfido to 2.1.2 (#88343)

balloob committed:
Fix dangling task for crownstone (#88292)

balloob committed:
Fix dangling task for mysensors (#88290)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Fix dangling task for insteon (#88293)

balloob committed:
Fix dangling task for smart_meter_texas (#88297)

balloob committed:
Fix dangling task for sense (#88296)

balloob committed:
Fix dangling task for plum_lightpad (#88295)

balloob committed:
Fix dangling task for squeezebox (#88299)

balloob committed:
Fix dangling task for livisi (#88294)

balloob committed:
Fix dangling task for bluetooth_le_tracker (#88284)

bdraco committed:
Dismiss discoveries when the underlying device disappears (#88340)

* Implement discovery removals

Bluetooth, HomeKit, SSDP, and Zeroconf now implement
dismissing discoveries when the underlying discovered
device disappears

* cover

* add zeroconf test

* cover

* cover bluetooth

* fix rediscover

synesthesiam committed:
Only expose default cloud domains in Assist default agent (#88274)

* Only expose default cloud domains in default agent

* Copy exposed domain list to conversation

* Implement requested changes

* Add test for exposed devices/areas

frenck committed:
Add support for snapshot testing (#88323)

* Add support for snapshot testing

* Use snapshots in Elgato diagnostics test

* Use snapshots in Elgato sensor test

* Fix flake8 warning

* Slightly improve serialized output

* Remove snapshot naming in elgato sensor tests

* Improve snapshoting of via_device_id in device registry item

* Update pylint typehints plugin for snapshot fixture typing

* Use snapshots in Elgato configflow test

* Use snapshots in Bluetooth repair issue tests

ludeeus committed:
Do not try to run DB jobs if there are no target groups (#88352)

ludeeus committed:
Use ubuntu-20.04 for Ruff lint job (#88355)

frenck committed:
Hold strong references to tasks in WLED (#88257)

AlePerla committed:
Removed integrated API for published API
Changed references to published API

AlePerla committed:
Merge remote-tracking branch 'upstream/dev' into zcs_azzurro_integration

AlePerla committed:
Updated icons and names
Skip update on connection errors

AlePerla committed:
Divided errors depending by the status code

AlePerla committed:
Merge branch 'dev' into zcs_azzurro_integration

AlePerla committed:
Merge branch 'dev' into zcs_azzurro_integration

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/__init__.py:
```diff
@@ -0,0 +1,49 @@
+    """Error to indicate there was a not ok response."""
+
+
+class ConfigEntryNotReady(HomeAssistantError):
+    """Error to indicate the entry is unreachable."""
+
+
+class ConfigEntryAuthFailed(HomeAssistantError):
+    """Error to indicate the auth to the entry failed."""
```
You shouldn't be creating them, you should be importing them
from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/__init__.py:
```diff
@@ -0,0 +1,49 @@
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up zcs_azzurro from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
```
you need to set hass data [entry] like you do in your async_setup_entry in sensor.

But you need to do it here, not in sensor.

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/sensor.py:
```diff
@@ -0,0 +1,342 @@
+    available_realtime_keys = list(realtime_sensors_available.keys())
+    _LOGGER.debug("available realtime keys are %s", ", ".join(available_realtime_keys))
+
+    hass.data[DOMAIN][entry.entry_id] = Inverter(
+        client=entry.data[SCHEMA_CLIENT_KEY],
+        thing_serial=entry.data[SCHEMA_THINGS_KEY],
+        name=entry.data[SCHEMA_FRIENDLY_NAME],
+    )
```
This should be done in init.

Lash-L commented on homeassistant/components/zcs_azzurro_inverter/config_flow.py:
```diff
@@ -0,0 +1,134 @@
+            SCHEMA_FRIENDLY_NAME: "Friendly name",
+        }
+
+        errors = {}
```
errors should be typed

AlePerla replied:
Should be solved now

Lash-L reviewed (commented):
A few quick things

AlePerla committed:
Imported correct errors instead of creating them

AlePerla committed:
Fixed requested changes

AlePerla committed:
Merge branch 'dev' into zcs_azzurro_integration

AlePerla commented:
@Lash-L I should have done all required changes

AlePerla committed:
Merge branch 'dev' into zcs_azzurro_integration

AlePerla committed:
Moved ConfigEntryAuthFailed and ConfigEntryNotReady only in init

AlePerla committed:
Merge branch 'dev' into zcs_azzurro_integration

AlePerla committed:
Fixed wrong placeholder left

AlePerla committed:
Tests have been fixed properly and executed

AlePerla commented:
Hi @Lash-L i did a mistake with tests and ran the wrong ones, i fixed them now, my apologies. Now they ran correctly

Lash-L commented:
@AlePerla 

You need to add your documentation PR to the description of this PR, it looks like you deleted the place it goes though? Copy the template fresh and update your description so you don't have the docs-missing workflow as failed.

AlePerla commented:
@Lash-L
The PR is still open ([https://github.com/home-assistant/home-assistant.io/pull/26298](https://github.com/home-assistant/home-assistant.io/pull/26298)) the one I closed is for the logo, since I haven't been authorised by the product owner to use their logo and the logo PR has been refused, so I don't know if the documentation will ever be accepted.
I will put it in the description anyway

AlePerla committed:
Merge branch 'dev' into zcs_azzurro_integration

Lash-L commented:
> @Lash-L
> 
> The PR is still open ([https://github.com/home-assistant/home-assistant.io/pull/26298](https://github.com/home-assistant/home-assistant.io/pull/26298)) the one I closed is for the logo, since I haven't been authorised by the product owner to use their logo and the logo PR has been refused, so I don't know if the documentation will ever be accepted.
> 
> I will put it in the description anyway

If you can't get your brands integration approved, this pr won't end up getting approved. So you should get that sorted. Otherwise this wont be approved by maintainers.
